src/components/number/number.test.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { createRef } from 'react';
+import {
+  render,
+  screen,
+  cleanup,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Number from './number.jsx';
+
+describe('Number', () => {
+  const requiredProps = {
+    id: 'test-component',
+    label: 'Test Component',
+  };
+
+  afterEach(cleanup);
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Number {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('disabled API', () => {
+    it('permits disabling', () => {
+      render(<Number {...requiredProps} disabled />);
+      expect(screen.getByLabelText('Test Component')).toBeDisabled();
+    });
+  });
+
+  describe('label API', () => {
+    it('presents a label', () => {
+      render(<Number {...requiredProps} />);
```

Comment:
We need to use the label to ensure we're asserting some implementation detail. 🤔